sycl/unittests/program_manager/passing_link_and_compile_options.cpp

```diff
@@ -0,0 +1,276 @@
+  EXPECT_EQ(expected_compile_options, current_compile_options);
+}
+
+TEST(Link_Compile_Options, one_compile_option_Test) {
+  sycl::platform Plt{sycl::default_selector()};
+  if (Plt.is_host()) {
+    std::cerr << "Test is not supported on host, skipping\n";
+    return; // test is not supported on host.
+  }
+
+  if (Plt.get_backend() == sycl::backend::ext_oneapi_cuda) {
+    std::cerr << "Test is not supported on CUDA platform, skipping\n";
+    return;
+  }
+
+  if (Plt.get_backend() == sycl::backend::ext_oneapi_hip) {
+    std::cerr << "Test is not supported on HIP platform, skipping\n";
+    return;
+  }
+  sycl::unittest::PiMock Mock{Plt};
+  setupDefaultMockAPIs(Mock);
+  Mock.redefine<sycl::detail::PiApiKind::piProgramCompile>(
+      redefinedProgramCompile);
+  Mock.redefine<sycl::detail::PiApiKind::piProgramLink>(redefinedProgramLink);
+
+  const sycl::device Dev = Plt.get_devices()[0];
+
+  current_link_options.clear();
+  current_compile_options.clear();
+  std::string expected_compile_options = "-cl-single-precision-constant";
+  std::string expected_link_options = "";
+  static sycl::unittest::PiImage DevImage =
+      generateEAMTestKernelImage<EAMTestKernel>(expected_compile_options,
+                                                expected_link_options);
+  static sycl::unittest::PiImageArray<1> DevImageArray{&DevImage};
+  auto KernelID = sycl::get_kernel_id<EAMTestKernel>();
+
+  sycl::queue Queue{Dev};
+
+  const sycl::context Ctx = Queue.get_context();
+  sycl::kernel_bundle KernelBundle =
+      sycl::get_kernel_bundle<sycl::bundle_state::input>(Ctx, {Dev},
+                                                         {KernelID});
+
+  auto BundleObj = sycl::compile(KernelBundle);
+  sycl::link(BundleObj);
+  EXPECT_EQ(expected_link_options, current_link_options);
+  EXPECT_EQ(expected_compile_options, current_compile_options);
+}
+
+TEST(Link_Compile_Options, one_link_and_compile_option_Test) {
+  sycl::platform Plt{sycl::default_selector()};
+  if (Plt.is_host()) {
+    std::cerr << "Test is not supported on host, skipping\n";
+    return; // test is not supported on host.
+  }
+
+  if (Plt.get_backend() == sycl::backend::ext_oneapi_cuda) {
+    std::cerr << "Test is not supported on CUDA platform, skipping\n";
+    return;
+  }
+
+  if (Plt.get_backend() == sycl::backend::ext_oneapi_hip) {
+    std::cerr << "Test is not supported on HIP platform, skipping\n";
+    return;
+  }
+  sycl::unittest::PiMock Mock{Plt};
+  setupDefaultMockAPIs(Mock);
+  Mock.redefine<sycl::detail::PiApiKind::piProgramCompile>(
+      redefinedProgramCompile);
+  Mock.redefine<sycl::detail::PiApiKind::piProgramLink>(redefinedProgramLink);
+
+  const sycl::device Dev = Plt.get_devices()[0];
+
+  current_link_options.clear();
+  current_compile_options.clear();
+  std::string expected_compile_options = "-cl-single-precision-constant";
+  std::string expected_link_options = "-cl-finite-math-only";
```

Comment:
1. Can we also test a case when there are multiple images with their own link options are merged together?
2. It would be also nice to add similar test for `sycl::build` API while we are here.

Reply:
Yes, sure. I'll add such kind of tests.

Reply:
Thanks